ServiceWebsite/ServiceWebsite/ClientApp/src/app/services/user-media.service.ts

```diff
@@ -58,11 +58,28 @@ export class UserMediaService extends MediaService {
     if (!this._navigator.mediaDevices || !this._navigator.mediaDevices.enumerateDevices) {
       throw new Error('enumerateDevices() not supported.');
     }
-    let updatedDevices: MediaDeviceInfo[] = await this._navigator.mediaDevices.enumerateDevices();
+
+    let updatedDevices: MediaDeviceInfo[];
+
+    const stream = await navigator.mediaDevices.getUserMedia({ audio: true, video: true });
+    if (stream.getVideoTracks().length > 0 && stream.getAudioTracks().length > 0) {
+      updatedDevices = await navigator.mediaDevices.enumerateDevices();
+    }
+
     updatedDevices = updatedDevices.filter(x => x.deviceId !== 'default' && x.kind !== 'audiooutput');
+
     this.availableDeviceList = Array.from(updatedDevices, device =>
       new UserMediaDevice(device.label, device.deviceId, device.kind, device.groupId)
     );
+
+    if (!stream) {
+      return;
```

Comment:
@kkodalikk  There is already a code in one of the service or function to close the stream.

Reply:
Made changes